packages/alchemy/src/provider/base.ts

```diff
@@ -30,19 +29,14 @@ export class AlchemyProvider extends SmartAccountProvider<HttpTransport> {
     const config = AlchemyProviderConfigSchema.parse(config_);
 
     const { chain, entryPointAddress, opts, ...connectionConfig } = config;
-    const _chain =
-      typeof chain === "number" ? SupportedChains.get(chain) : chain;
```

Comment:
this is alchemy provider. don't we still need this as we have a notion of supported chains for alchemy?

Reply:
correct. now these checks are in the `AlchemyChainSchema` which is called in `AlchemyProviderConfigSchema.parse` on line 30.